modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoveryTopologyChangeAndReconnectTest.java

```diff
@@ -62,11 +66,35 @@
  * Tests for Zookeeper SPI discovery.
  */
 public class ZookeeperDiscoveryTopologyChangeAndReconnectTest extends ZookeeperDiscoverySpiTestBase {
+    /** {@code True} if indexing disabled. */
+    private boolean indexingDisabled;
+
     /** {@inheritDoc} */
     @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
-        return super.getConfiguration(igniteInstanceName).setIncludeEventTypes(EventType.EVTS_ALL);
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setIncludeEventTypes(EventType.EVTS_ALL);
+
+        if (indexingDisabled) {
+            GridQueryProcessor.idxCls = DummyQueryIndexing.class;
+
+            cfg.setSystemViewExporterSpi(new JmxSystemViewExporterSpi() {
```

Comment:
Let's fix `SqlViewExporterSpi` instead of excluding it (looks like a bug in its implementation)

Reply:
@alex-plekhanov, I have fixed it. Take a look, please. [Tests OK.](https://ci.ignite.apache.org/viewLog.html?buildId=4714492&buildTypeId=IgniteTests24Java8_ZooKeeperDiscovery1)